ast/ddl.go

```diff
@@ -1048,6 +1050,103 @@ type TableOption struct {
 	UintValue uint64
 }
 
+func (n *TableOption) Restore(ctx *RestoreCtx) error {
+	switch n.Tp {
+	case TableOptionEngine:
+		ctx.WriteKeyWord("ENGINE ")
+		ctx.WritePlain("= ")
+		if n.StrValue != "" {
+			ctx.WritePlain(n.StrValue)
+		} else {
+			ctx.WritePlain("''")
+		}
+	case TableOptionCharset:
+		ctx.WriteKeyWord("DEFAULT CHARACTER SET ")
+		ctx.WritePlain("= ")
+		ctx.WriteKeyWord(n.StrValue)
+	case TableOptionCollate:
+		ctx.WriteKeyWord("DEFAULT COLLATE ")
+		ctx.WritePlain("= ")
+		ctx.WriteKeyWord(n.StrValue)
+	case TableOptionAutoIncrement:
+		ctx.WriteKeyWord("AUTO_INCREMENT ")
+		ctx.WritePlain("= ")
+		ctx.WritePlainf("%d", n.UintValue)
+	case TableOptionComment:
+		ctx.WriteKeyWord("COMMENT ")
+		ctx.WritePlain("= ")
+		ctx.WritePlainf("'%s'", n.StrValue)
```

Comment:
No this should be `WriteString` not `WritePlainf`, the quotation marks need to be properly escaped (the same for all other `WritePlainf("'%s'", stuff)`.

```sql
ALTER TABLE foo COMMENT = 'abc\ndef''ghi';
```